perfmetrics/scripts/generate_files.py

```diff
@@ -41,29 +61,46 @@ def generate_files_and_upload_to_gcs_bucket(destination_blob_name, num_of_files,
           out.truncate(1024 * 1024 * int(file_size))
         if(file_size_unit.lower() == 'kb'):
           out.truncate(1024 * int(file_size))
+        if(file_size_unit.lower() == 'b'):
+          out.truncate(int(file_size))
+
+    num_files = os.listdir(TEMPORARY_DIRECTORY)
+    if not num_files:
+      return 0
 
     # Uploading batch files to GCS
-    process = Popen(
-        'gsutil -m cp -r {}/* {}'.format(TEMPORARY_DIRECTORY,
-                                         destination_blob_name),
-        shell=True)
-    process.communicate()
-    exit_code = process.wait()
-    if(exit_code != 0):
-      print('Exited with code {}'.format(exit_code))
-      subprocess.call('bash', shell=True)
+    if upload_to_gcs_bucket:
+      process = Popen(
+          'gsutil -m cp -r {}/* {}'.format(TEMPORARY_DIRECTORY,
+                                          destination_blob_name),
+          shell=True)
+      process.communicate()
+
+      exit_code = process.wait()
+      if(exit_code != 0):
+        if logging_method == 'output_file':
```

Comment:
please avoid taking logging_method as input param. you can return the exit_code here in both cases and handle the abort in the main method.

Reply:
done